src/storage/mutate/AddEdgesAtomicProcessor.cpp

```diff
@@ -0,0 +1,139 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include <folly/container/Enumerate.h>
+#include <algorithm>
+
+#include "codec/RowWriterV2.h"
+#include "storage/mutate/AddEdgesAtomicProcessor.h"
+#include "storage/transaction/TransactionManager.h"
+#include "storage/transaction/TransactionUtils.h"
+#include "utils/IndexKeyUtils.h"
+#include "utils/NebulaKeyUtils.h"
+
+#include <folly/executors/QueuedImmediateExecutor.h>
+
+namespace nebula {
+namespace storage {
+
+using ChainId = std::pair<PartitionID, PartitionID>;
+
+void AddEdgesAtomicProcessor::process(const cpp2::AddEdgesRequest& req) {
+    propNames_ = req.get_prop_names();
+    spaceId_ = req.get_space_id();
+
+    auto stVidLen = env_->schemaMan_->getSpaceVidLen(spaceId_);
+    if (!stVidLen.ok()) {
+        LOG(ERROR) << stVidLen.status();
+        for (auto& part : req.get_parts()) {
+            pushResultCode(cpp2::ErrorCode::E_INVALID_SPACEVIDLEN, part.first);
+        }
+        onFinished();
+        return;
+    }
+    vIdLen_ = stVidLen.value();
+    processByChain(req);
+}
+
+void AddEdgesAtomicProcessor::processByChain(const cpp2::AddEdgesRequest& req) {
+    auto ver = FLAGS_enable_multi_versions
+                       ? std::numeric_limits<int64_t>::max() - time::WallClock::fastNowInMicroSec()
+                       : FLAGS_default_mvcc_ver;
+    std::unordered_map<ChainId, std::vector<KV>> edgesByChain;
+    // split req into chains
+    bool invalidRequest = false;
+    for (auto& part : req.parts) {
+        for (auto& edge : part.second) {
+            auto& localPart = part.first;
+            auto stPartId = env_->metaClient_->partId(spaceId_, edge.key.dst.getStr());
+            if (!stPartId.ok()) {
+                invalidRequest = true;
+                pushResultCode(cpp2::ErrorCode::E_SPACE_NOT_FOUND, localPart);
```

Comment:
E_PART_NOT_FOUND ?

Reply:
I know it's wired, but env_->metaClient_->partId only return  Status"space not found" , that's why I use E_SPACE_NOT_FOUND.